text/0066-separate-context-propagation.md

```diff
@@ -0,0 +1,557 @@
+# Context Propagation: A Layered Approach 
+
+* [Motivation](#Motivation)
+* [OpenTelemetry layered architecture](#OpenTelemetry-layered-architecture)
+  * [Cross-Cutting Concerns](#Cross-Cutting-Concerns)
+    * [Observability API](#Observability-API)
+    * [Correlations API](#Correlations-API)
+  * [Context Propagation](#Context-Propagation)
+    * [Context API](#Context-API)
+    * [Propagation API](#Propagation-API)
+* [Prototypes](#Prototypes)
+* [Examples](#Examples)
+  * [Global initialization](#Global-initialization)
+  * [Extracting and injecting from HTTP headers](#Extracting-and-injecting-from-HTTP-headers)
+  * [Simplify the API with automated context propagation](#Simplify-the-API-with-automated-context-propagation)
+  * [Implementing a propagator](#Implementing-a-propagator)
+  * [Implementing a concern](#Implementing-a-concern)
+  * [The scope of current context](#The-scope-of-current-context)
+  * [Referencing multiple contexts](#Referencing-multiple-contexts)
+  * [Falling back to explicit contexts](#Falling-back-to-explicit-contexts)
+* [Internal details](#Internal-details)
+* [FAQ](#faq)
+
+Refactor OpenTelemetry into a set of separate cross-cutting concerns which 
+operate on a shared context propagation mechanism.
+
+# Motivation
+
+This RFC addresses the following topics:
+
+**Separation of concerns**  
+* Remove the Tracer dependency from context propagation mechanisms.
+* Handle user data (Correlations) and observability data (SpanContext, etc) 
+  separately.  
+
+**Extensibility**
```

Comment:
I remember other motivation factors were things like context-based app and resource definition. Is it out of scope?

Reply:
I am leaving that out of scope here. Those changes are more like a static or lexical scope: how to apply the same set of resources to every "event" which occurs in a certain "region." This is focused solely on how to track and propagate execution context. 

@jmacd and @danielkhan (and @bogdandrutu I think) are working on the resource/name-spacing issue separately.

Reply:
trace context is one of the scopes for a resource. I want to make sure that justification covers this aspect.